File: library/src/main/java/me/proxer/library/entity/manga/Page.java
package me.proxer.library.entity.manga;

import lombok.Value;

/**
 * Entity representing a single page from a {@link Chapter}.
 *
 * @author Ruben Gees
 */
@Value
public class Page {

    /**
     * Returns the name of the page. To be used for retrieving the url.
     */
    private String name;

    /**
     * Returns the height of the page.
     */
    private int height;

    /**
     * Returns the width of the uploader if present.
     */
    private int width;
}

File: library/src/main/kotlin/me/proxer/library/entity/manga/Page.kt
package me.proxer.library.entity.manga

import com.squareup.moshi.Json

/**
 * Entity representing a single page from a [Chapter].
 *
 * @property name The name of the page. To be used for retrieving the url.
 * @property height The height of the page.
 * @property width The width of the page.
 *
 * @author Ruben Gees
 */
data class Page(
    @Json(name = "name") val name: String,
    @Json(name = "height") val height: Int,
    @Json(name = "width") val width: Int
)
